e2etest/base/main.py

```diff
@@ -94,7 +95,7 @@ def main(cb_loop: asyncio.AbstractEventLoop, test_cfg: TestCfg):
     """
     Generate the device and run the end to end tests.
     """
-    persistency_dir = Path.joinpath(Path.cwd(), "e2etest", "base", "build")
+    persistency_dir = Path(tempfile.TemporaryDirectory("astarte-sdk-python-e2e").name)
     if not Path.is_dir(persistency_dir):
         os.makedirs(persistency_dir)
```

Comment:
It can be used in the same manner as when opening a file. No need to check if the directory exists.
```suggestion
    with tempfile.TemporaryDirectory(prefix="astarte-sdk-python-e2e") as persistency_dir:
```